components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go

```diff
@@ -563,6 +563,110 @@ func TestReconciler_APIRuleConfig_Upgrade(t *testing.T) {
+
+	var testCases = []struct {
+		name                   string
+		givenSubscription      *eventingv1alpha2.Subscription
+		givenReconcilerSetup   func(*eventingv1alpha2.Subscription) (*Reconciler, client.Client)
+		wantEv2Hash            int64
+		wantEventMeshHash      int64
+		wantWebhookAuthHash    int64
+		wantEventMeshLocalHash int64
+		wantReconcileError     error
+	}{
+		{
+			name: "Preserve hashes if conditions are empty",
+			givenSubscription: func() *eventingv1alpha2.Subscription {
+				return reconcilertesting.NewSubscription("some-test-sub-0", "test",
+					reconcilertesting.WithValidSink("test", "some-test-svc-0"),
+					reconcilertesting.WithConditions(nil),
+					reconcilertesting.WithBackend(eventingv1alpha2.Backend{
+						Ev2hash:            int64(4118518533334734626),
+						EventMeshHash:      int64(4748405436686967274),
+						WebhookAuthHash:    int64(4118518533334734627),
+						EventMeshLocalHash: int64(4883494500014499539),
+					}),
+				)
+			}(),
+			givenReconcilerSetup: func(s *eventingv1alpha2.Subscription) (*Reconciler, client.Client) {
+				te := setupTestEnvironment(t, s)
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("SyncSubscription", mock.Anything, mock.Anything, mock.Anything).Return(true, nil)
+				return NewReconciler(ctx, te.fakeClient, te.logger, te.recorder, te.cfg, te.cleaner,
+					te.backend, te.credentials, te.mapper, validator, collector), te.fakeClient
+			},
+			wantEv2Hash:            int64(4118518533334734626),
+			wantEventMeshHash:      int64(4748405436686967274),
+			wantWebhookAuthHash:    int64(4118518533334734627),
+			wantEventMeshLocalHash: int64(4883494500014499539),
+			wantReconcileError:     nil,
+		},
+		{
+			name: "Preserve hashes if conditions are not empty",
+			givenSubscription: func() *eventingv1alpha2.Subscription {
+				return reconcilertesting.NewSubscription("some-test-sub-1", "test",
+					reconcilertesting.WithValidSink("test", "some-test-svc-1"),
+					reconcilertesting.WithConditions(eventingv1alpha2.MakeSubscriptionConditions()),
+					reconcilertesting.WithBackend(eventingv1alpha2.Backend{
+						Ev2hash:            int64(2118518533334734626),
+						EventMeshHash:      int64(2748405436686967274),
+						WebhookAuthHash:    int64(2118518533334734627),
+						EventMeshLocalHash: int64(2883494500014499539),
+					}),
+				)
+			}(),
+			givenReconcilerSetup: func(s *eventingv1alpha2.Subscription) (*Reconciler, client.Client) {
+				te := setupTestEnvironment(t, s)
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("SyncSubscription", mock.Anything, mock.Anything, mock.Anything).Return(true, nil)
+				return NewReconciler(ctx, te.fakeClient, te.logger, te.recorder, te.cfg, te.cleaner,
+					te.backend, te.credentials, te.mapper, validator, collector), te.fakeClient
+			},
+			wantEv2Hash:            int64(2118518533334734626),
```

Comment:
Can we extract constants for tests like these. Its really hard to read those random numbers and nobody should actually think (s)he has to compare/understand those values

Reply:
Done